### PR TITLE
Allow us to monitor dodgy user UUIDs

### PR DESCRIPTION
**Pull request by keirog**
We occasionally get missing/malformed user UUIDs in events that means we can’t determine who is trying to withdraw consent.

Behavior change in this PR:
- stop hitting the consent store if user UUID is malformed
- for events with malformed user UUIDs, log warning to Splunk
(after merge: - add alert to Splunk so we’re informed promptly and we can stop it from clogging up the queue)

### CODE REVIEW

File: secret-squirrel.js [hunk 1]
```diff
@@ -1,67 +1,14 @@
 module.exports = {
 	files: {
 		allow: [
-			'.jshintrc'
+			'.jshintrc',
+			'package-lock.json'
```
Comment (AbdullahAli) [inline secret-squirrel.js]: ohhh yeh!!
```
🤘🏼
  \ 👀
    👃🏼
    👅
```

File: tests/app/services/consent-store.test.js [hunk 2]
```diff
@@ -87,14 +91,21 @@ describe('Withdrawing consent via the SCS:', () => {
...
 			.then(() => done(new Error('This should’ve rejected.')))
 			.catch(() => done());
 	});
 
+	it('should reject user UUIDs that don’t match the valid UUID v4 length (36)', () => {
+		const clonedMockData = JSON.parse(JSON.stringify(mockGoodData));
+
+		clonedMockData.context.userUuid = badUuid;
+
+		return expect(consentStore.withdrawConsent(clonedMockData)).to.be.rejected;
```
Comment (AbdullahAli) [inline tests/app/services/consent-store.test.js]: 👍🏼

Comment (AbdullahAli) [reply tests/app/services/consent-store.test.js]: Would be useful to surface the expected error message here using [rejectedWith](https://www.chaijs.com/plugins/chai-as-promised/#multiple-promise-assertions)

Comment (keirog) [reply tests/app/services/consent-store.test.js]: 👍 Good shout

**Pull request merged**